api/run.py

```diff
@@ -609,5 +609,27 @@ def changeGrades():
     studentsDOM.changeGrades(int(request.json['difference']))
     return '0'
 
+@app.route('/dataDownload', methods = ['POST'])
+@requires_auth
+@log_action('data download')
+@specific_roles(['admin', 'developer'])
+def dataDownload():
+    toDownload = request.json['toDownload']
```

Comment:
Could have been modular but subprocess was not happy with a variable in the string. :(